covid19-dashboard/main.py

```diff
@@ -12,201 +12,488 @@
+
+    state_names = _get_place_names(state_geo_ids)
+
+    # Store all the US State metadata as a dict of tuples.
+    # geo_id -> (name, containedIn).
+    states: PlaceToInfoType = {
+        geo_id: (name, "country/USA")
+        for geo_id, name in state_names.items()
+    }
+
+    # Get US County data belonging to the US States.
+    response = send_request(
+        DC_SERVER + "node/places-in",
+        {"dcids": state_geo_ids,
+         "placeType": "County"},
+        api_key=API_KEY,
+        )
+
+    # Keep track of what State each County belongs to.
+    # geo_id -> belongs_to_geo_id.
+    # EXAMPLE: {"geoId/12000": "geoId/12"}
+    county_to_state: Dict[str, str] = {}
     for value in response:
-        if 'place' in value and 'dcid' in value:
-            county_geoId: str = value['place']
-            belongs_to_state_geoId: str = value['dcid']
-            county_to_state[county_geoId]: str = belongs_to_state_geoId
-
-    # Get the name of all counties
-    response: dict = send_request(DC_SERVER + "node/property-values", {
-        "dcids": list(county_to_state.keys()),
-        "property": "name"
-    },
-                                  api_key=API_KEY)
-
-    # Store all the counties
-    counties: dict = {}
-    for geoId in response:
-        if 'out' in response[geoId] and response[geoId][
-                'out'] and 'value' in response[geoId]['out'][0]:
-            name: str = response[geoId]['out'][0]['value']
-            belongs_to = county_to_state[geoId]
-            counties[geoId] = (name, belongs_to)
-
-    counties['geoId/3651000'] = ("New York City", "geoId/36")
-    counties['geoId/2938000'] = ("Kansas City", "geoId/29")
-    return {**states, **counties}
-
-
-def get_stats_by_date(place: list, stats_var: str) -> dict:
-    """
-    Queries and returns the covid data in the appropriate format.
-    :param place: list of dcids to get data for
-    :param stats_var: the Data Commons stats_var to query data for
-    :return: dictionary where every day contains several regions,
-    and those regions contain an int of cases/deaths
-    """
-    response: dict = send_request(DC_SERVER + "bulk/stats", {
-        "place": place,
-        "stats_var": stats_var
-    },
-                                  api_key=API_KEY)
-    output: defaultdict = defaultdict(dict)
-
-    # Store the data in output as dates->dcid->value
-    for dcid in response:
-        if not response[dcid]:
+        if "place" not in value:
             continue
-        if "data" not in response[dcid]:
+        if "dcid" not in value:
             continue
-        data: dict = response[dcid]["data"]
+
+        county_geo_id: str = value["place"]
+        belongs_to_geo_id: str = value["dcid"]
+        county_to_state[county_geo_id]: str = belongs_to_geo_id
+
+    # If the user didn't request County data, they requested State data.
+    # Return state data.
+    if place_type != "County":
```

Comment:
this statement can be moved up to save some computation.

Reply:
Fixed, thanks!